### PR TITLE
Add clear button to the search bar of Configuration side panel

### PR DESCRIPTION
**Pull request by MatevzFa**
Preview:
java_2018-07-23_20-29-33

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package net.runelite.client.plugins.config;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java]: Add license to each new file.

File: runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.awt.Dimension;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import javax.swing.ImageIcon;
+import net.runelite.client.ui.ColorScheme;
+import net.runelite.client.ui.PluginPanel;
+import net.runelite.client.ui.components.IconButton;
+import net.runelite.client.ui.components.IconTextField;
+
+public class ConfigSearchBar extends IconTextField
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java]: This probably should not be `public`

File: runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	public ConfigSearchBar()
+	{
+		super();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigSearchBar.java]: no-arg super calls are implicitly generated by the compiler, and we generally omit them

**Comment by MatevzFa** [comment]
The client didn't load for me on this branch until I synced it. Should I push that?

**Comment by MagicfTail** [comment]
This was added at some point

**Comment by MagicfTail** [comment]
#3451 I think

**Pull request closed without merging**